study1/week5/fair-candy-swap.java

```diff
@@ -0,0 +1,34 @@
+//https://leetcode.com/problems/fair-candy-swap/description/
+
+class Solution {
+    public int[] fairCandySwap(int[] aliceSizes, int[] bobSizes) {
+        //time complexity is O(n*m),where n is the size of aliceSizes and m is the size of bobSizes
+        //space complexity is O(1)
+        int aSum = sum(aliceSizes);
+        int bSum = sum(bobSizes);
+        int mid = (aSum + bSum) / 2;
+        int diff = bSum - mid;
+
+        int[] result = new int[2];
+        Arrays.sort(aliceSizes);
+        Arrays.sort(bobSizes);
+        
+        for (int aNum : aliceSizes) {
+            for (int bNum : bobSizes) {
```

Comment:
If you store bobSizes in a set, you can query this whole loop with a single `bobSizesSet.has(aNum + diff)` then this loop becomes `O(n)`